apps/nextra/pages/en/network/nodes/validator-node/modify-nodes/rotate-consensus-key.mdx

```diff
@@ -0,0 +1,135 @@
+---
+title: "Rotate consensus key"
+---
+
+import { Callout, Steps } from 'nextra/components';
+
+# Rotate consensus key
+
+Consensus key rotation is an action taken by a node operator to change the identity of a validator they control.
+It happens typically when the corresponding private key is (potentially) lost/leaked,
+or every X months as a common security practice.
+
+Below is the step-by-step guide of performing a consensus key rotation with detailed examples.
+
+## New key generation
+
+Using the following [aptos CLI](https://aptos.dev/en/build/cli) command to generate a new consensus identity.
+In this example, identity files are saved to directory `/new/key/root`.
+```bash
+aptos genesis generate-keys --output-dir /new/key/root
+```
+
+## Add new private key to node
+
+Edit the node config yaml as described below.
+```yaml
+# ...
+consensus:
+    # ...
+    safety_rules:
+        # ...
+        initial_safety_rules_config:
+            from_file:
+                # ...
+                identity_blob_path: /old/key/root/validator-identity.yaml
+                overriding_identity_paths:              # new!
+                - /new/key/root/validator-identity.yaml # new!
+```
+<Callout type="info">
+Don't delete `identity_blob_path`. It is a required field.
+
+Field `overriding_identity_paths` is optional, and allows multiple identities to be specified.
+
+Among all the identities, the node finds the one that appears on-chain validator set, and uses it as the consensus key.
+</Callout>
+
+<Callout type="info">
+Old keys may still be available even if you update `identity_blob_path`
+or delete items from `overriding_identity_paths`,
+due to the reasons below.
+
+There is a separate **secure storage** (typically a file `secure_storage.json`) hosting all identities.
+The content of `identity_blob_path` and `overriding_identity_paths` are added to the secure storage on node start, if missing.
+However, keys are never deleted automatically from the storage.
+
+To completely wipe out an old key, you will have to manually make changes to the secure storage.
+See the end of the guide for more details.
+</Callout>
```

Comment:
Why are there two callouts?

I would just remove the callouts altogether

Reply:
deleted this callout, and only kept the first one.